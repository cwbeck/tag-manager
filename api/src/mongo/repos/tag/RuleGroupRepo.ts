import { injectable } from 'inversify';
import RuleGroup from '../../models/tag/RuleGroup';
import UnderRevisionControl from './abstractions/UnderRevisionControl';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class RuleGroupRepo extends UnderRevisionControl<RuleGroup> {
    protected readonly auditEnabled = true;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                ___persisting_id: 1,
                _revision_id: 1,
            },
            unique: true,
        },
    ];
}
