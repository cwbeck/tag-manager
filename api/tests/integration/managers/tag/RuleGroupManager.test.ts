import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import RuleGroupRepo from '../../../../src/mongo/repos/tag/RuleGroupRepo';
import Mock = jest.Mock;

describe('RuleGroupManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches a single rule group', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();

        const mockSutModel = TestUtils.buildMockRuleGroup();
        const mockSutRepo = {} as RuleGroupRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getRuleGroup(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    name
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
