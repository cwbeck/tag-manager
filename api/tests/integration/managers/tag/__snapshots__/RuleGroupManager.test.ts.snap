// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleGroupManager fetches a single rule group 1`] = `
Object {
  "data": Object {
    "getRuleGroup": Object {
      "id": "5edf4c48e9615d9a1c66bbff",
      "name": "test rule group",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
